t/porting/authors.t

```diff
@@ -19,6 +19,10 @@ skip_all(
     "This is a shallow clone, this test requires history.")
   if (-e "$source_dir/.git/shallow");
 
+skip_all(
+    "Cannot locate .mailmap: GH#21272.")
+  unless (-f ".mailmap");
+
```

Comment:
Agree, this can be fixed instead of skipped

Reply:
> Agreed, this can be fixed instead of skipped

Okay, but since I was skeptical of the work put into Porting/updateAUTHORS.p[lm] in the last dev cycle, someone other than me should provide the fix.  Have at it!

Reply:
I'll pick it up.

Reply:
@Leont fwiw, I don't really agree. The authors.t is an authors test, it only needs to run when someone is using a git repo, and is doing a "standard" build in the git tree. We can make it work with -Dmksymlinks, but IMO that isn't a normal dev mode where someone would be expected to create patches from.  I am not sure if supporting this test really makes sense. I do have an alternative patch, but honestly I think @jkeenan's patch was fine. This test is only interesting when someone is actually writing patches.